### PR TITLE
[#454]On first run, start SurveyDownloadService after DeviceID is set

### PR DESCRIPTION
**Pull request by stellanl**

### CODE REVIEW

File: app/src/main/java/org/akvo/flow/activity/SurveyActivity.java [hunk 1]
```diff
@@ -127,12 +127,14 @@ public void onDrawerOpened(View drawerView) {
...
         // Start the setup Activity if necessary.
+        boolean noDevIdYet = false;
         if (!Prefs.getBoolean(this, Prefs.KEY_SETUP, false)) {
+            noDevIdYet = true;
             startActivityForResult(new Intent(this, AddUserActivity.class), REQUEST_ADD_USER);
         }
 
         displaySelectedUser();
-        startServices();
+        startServices(noDevIdYet);
```
Comment (mtwestra) [inline app/src/main/java/org/akvo/flow/activity/SurveyActivity.java]: Isn't noDevIdYet still true here? In that case, the service will not run?

Comment (stellanl) [inline app/src/main/java/org/akvo/flow/activity/SurveyActivity.java]: Yes, that is the idea. It will not be started here, but only on return from the get-user-and-devid screen.

File: app/src/main/java/org/akvo/flow/activity/AddUserActivity.java [hunk 2]
```diff
@@ -61,6 +63,10 @@ public void onClick(View v) {
                 db.savePreference(ConstantUtil.DEVICE_IDENT_KEY, deviceId);
                 db.close();
 
+                // Trigger the SurveyDownload Service, in order to make
+                // a backend connection with the new Device ID
```
Comment (mtwestra) [inline app/src/main/java/org/akvo/flow/activity/AddUserActivity.java]: Isn't it strange to trigger the surveydownload service within the adduseractivity? It seems to mix things that are unrelated?

Comment (stellanl) [inline app/src/main/java/org/akvo/flow/activity/AddUserActivity.java]: I guess it could be moved to where SurveyActivity receives the return value from AddUserActivity. I was sticking as close as possible to the similar code in PreferencesActivity.

Comment (mtwestra) [inline app/src/main/java/org/akvo/flow/activity/AddUserActivity.java]: I guess I would be in favour of moving it there, as it will clarify the intent and structure of the code.

**Comment by stellanl** [comment]
Should be merged into develop, not master
